README.md

```diff
@@ -1,39 +1,163 @@
+
+## (Highly) Recommended Reading
+
+Working in this repository is easier if you are familiar with Buildroot.
+
+- Documentation and users manual: <https://buildroot.org/docs.html>
+- Upstream repository: <https://github.com/buildroot/buildroot/>
+- Buildroot website: <https://buildroot.org/>
 
 ## Building Locally
 
-We wrap the buildroot build system in a docker container to manage dependencies and make it slightly more portable. People have had problems building this repo on things other than Linux, however, and that remains really the only supported build platform - but it probably works on other platforms too.
+We wrap the Buildroot build system in a [Docker][] container to manage dependencies and make it slightly more portable. Due to case-sensitive filenames and other concerns, the build must be run on Linux.
+
+This fork requires the [opentrons repo](https://github.com/Opentrons/opentrons) to be checked out next to it, since it uses a Buildroot external toolchain to build in our dependencies.
+
+[docker]: https://www.docker.com/
+
+### Prerequisites
+
+- Linux
+  - If you are on another OS, you should use a local or cloud Linux virtual machine
+  - A tool like [Vagrant][] can help with VM configuration and setup
+- `docker`
+- `git`
+
+[vagrant]: https://www.vagrantup.com/
+
+### Setup
+
+In order to build, `docker` and `git` must be installed and the monorepo must be checked out as a neighbor to `buildroot`
+
+```shell
+# 0) clone the opentrons monorepo, if you haven't already
+git clone https://github.com/Opentrons/opentrons.git
+
+# 1) clone the Opentrons fork of buildroot next to the monorepo
+git clone https://github.com/Opentrons/buildroot.git
+
+# 2) navigate into the repository
+cd buildroot
+```
+
+### Build
+
+From `buildroot`, run the `./opentrons_build.sh` script. This will build a docker container and run a build in it using the Buildroot `Makefile`
+
+```shell
+# build all images
+./opentrons_build.sh
+
+# equivalent command
+./opentrons_build.sh all
+```
+
+Buildroot caches build intermediates aggressively, but sometimes you need to run a full rebuild. Full rebuilds (as well as your very first builds) can take a long time, so avoid them if you can. The Buildroot docs explain [when a full rebuild is necessary][].
+
+```shell
+# run a full rebuild
+./opentrons_build.sh "clean all"
+```
+
+[when a full rebuild is necessary]: https://buildroot.org/downloads/manual/manual.html#full-rebuild
+
+### Outputs
 
-This fork requires the [opentrons repo](https://github.com/Opentrons/opentrons) to be checked out next to it, since it uses a buildroot external toolchain to build in our dependencies.
+The outputs of the build are
+
+- A system image, `output/images/ot2-system.zip`, good for doing a software update
+  - If your build is unsigned you will need to disable signature checking in the robot's update server config (`/var/lib/otupdate/config.json`) and restart it (`systemctl restart opentrons-update-server`).
```

Comment:
Add a note - you can tell the difference between signed and unsigned builds by looking at the files in `/output/images`. If there are signature files, the build is signed; if there are not, the build is unsigned. Most builds on your machine will be unsigned unless you provide the opentrons signing key to the build on the command line using the appropriate argument. You can provide any signing key to the build, but robots on signed versions will verify against the public key for the opentrons signing key.